compose/material/material/src/desktopMain/kotlin/androidx/compose/material/DesktopAlertDialog.desktop.kt

```diff
@@ -1,437 +0,0 @@
-    @Composable
-    override fun AlertDialog(
-        onDismissRequest: () -> Unit,
-        content: @Composable () -> Unit
-    ) {
-        AlertDialog(
-            onDismissRequest = onDismissRequest,
-            shape = RectangleShape,
-            modifier = Modifier,
-            content = { content() }
-        )
-    }
-
-    @Composable
-    override fun AlertDialog(
-        onDismissRequest: () -> Unit,
-        shape: Shape,
-        modifier: Modifier,
-        content: @Composable (Modifier) -> Unit
-    ) {
-        // Popups on the desktop are by default embedded in the component in which
-        // they are defined and aligned within its bounds. But an [AlertDialog] needs
-        // to be aligned within the window, not the parent component, so we cannot use
-        // [alignment] property of [Popup] and have to use [Box] that fills all the
-        // available space. Also [Box] provides a dismiss request feature when clicked
-        // outside of the [AlertDialog] content.
-        Popup(
-            popupPositionProvider = object : PopupPositionProvider {
-                override fun calculatePosition(
-                    anchorBounds: IntRect,
-                    windowSize: IntSize,
-                    layoutDirection: LayoutDirection,
-                    popupContentSize: IntSize
-                ): IntOffset = IntOffset.Zero
-            },
-            focusable = true,
-            onDismissRequest = onDismissRequest,
-            onKeyEvent = {
-                if (it.type == KeyEventType.KeyDown && it.awtEventOrNull?.keyCode == KeyEvent.VK_ESCAPE) {
-                    onDismissRequest()
-                    true
-                } else {
-                    false
-                }
-            },
-        ) {
-            val scrimColor = Color.Black.copy(alpha = 0.32f) //todo configure scrim color in function arguments
-            Box(
-                modifier = Modifier
-                    .fillMaxSize()
-                    .background(scrimColor)
-                    .pointerInput(onDismissRequest) {
-                        detectTapGestures(onPress = { onDismissRequest() })
-                    },
-                contentAlignment = Alignment.Center
-            ) {
-                content(
-                    modifier
-                        .shadow(elevation = 24.dp, shape = shape)
-                        .pointerInput(onDismissRequest) {
-                            detectTapGestures(onPress = {
-                                // Workaround to disable clicks on Surface background
-                                // https://github.com/JetBrains/compose-jb/issues/2581
-                            })
-                        }
-                )
-            }
-        }
-    }
-}
-
-/**
- * Shows Alert dialog as undecorated draggable window.
- */
-@ExperimentalMaterialApi
-object UndecoratedWindowAlertDialogProvider : AlertDialogProvider {
```

Comment:
Will it be hard to deprecate it first? It wasn't popular though:
https://kotlinlang.slack.com/archives/C01D6HTPATV/p1628776950275200?thread_ts=1628686951.262700&cid=C01D6HTPATV
https://kotlinlang.slack.com/archives/C01D6HTPATV/p1627907106077200?thread_ts=1627906634.076600&cid=C01D6HTPATV
https://github.com/search?q=UndecoratedWindowAlertDialogProvider&type=code

Reply:
> Will it be hard to deprecate it first?

I don't think that we need this for `@ExperimentalMaterialApi`